src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double MULTIPLIER_FOR_ARRAY_SIZE = 1.5;
+    private int arrayCapacity = 10;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[arrayCapacity];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= arrayCapacity) {
+            newArray();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("incorrect index");
+        } else if (index < size) {
+            shiftRightAndAddElement(value, index);
+            return;
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIfIndexCorrect(index);
+        return array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIfIndexCorrect(index);
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIfIndexCorrect(index);
+        T temp = array[index];
+        shiftLeftAndRemoveElement(index);
+        return temp;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if ((array[i] == element) || (array[i] != null && array[i].equals(element))) {
+                return remove(i);
+            }
+        }
+        throw new NoSuchElementException("no such element in list");
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size < 1;
+    }
+
+    private T[] newTemporaryArray() {
+        T[] tempArray = (T[]) new Object[arrayCapacity];
+        System.arraycopy(array, 0, tempArray, 0, array.length);
+        return tempArray;
+    }
+
+    private void newArray() {
+        arrayCapacity *= MULTIPLIER_FOR_ARRAY_SIZE;
+        array = newTemporaryArray();
+    }
```

Comment:
merge logic from these 2 methods and call it `resize`. Also, to get the current array size, you should use `array.length` instead of storing the array size in the class variable `arrayCapacity` and change it